src/domain/staking/contracts/decoders/kiln-decoder.helper.ts

```diff
@@ -0,0 +1,124 @@
+import { AbiDecoder } from '@/domain/contracts/decoders/abi-decoder.helper';
+import { LoggingService, ILoggingService } from '@/logging/logging.interface';
+import { Inject, Injectable } from '@nestjs/common';
+
+export const KilnAbi = [
+  {
+    inputs: [],
+    name: 'deposit',
+    outputs: [],
+    stateMutability: 'payable',
+    type: 'function',
+  },
+  {
+    inputs: [{ internalType: 'bytes', name: '_publicKeys', type: 'bytes' }],
+    name: 'requestValidatorsExit',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [{ internalType: 'bytes', name: '_publicKeys', type: 'bytes' }],
+    name: 'batchWithdrawCLFee',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+] as const;
+
+export type KilnRequestValidatorsExitParameters = {
+  name: '_publicKeys';
+  type: 'bytes';
+  value: `0x${string}`;
+  valueDecoded: null;
+};
+export type KilnBatchWithdrawCLFeeParameters =
+  KilnRequestValidatorsExitParameters;
+
+@Injectable()
+export class KilnDecoder extends AbiDecoder<typeof KilnAbi> {
+  constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+  ) {
+    super(KilnAbi);
+  }
+
+  decodeDeposit(
+    data: `0x${string}`,
+  ): { method: string; parameters: [] } | null {
+    if (!this.helpers.isDeposit(data)) {
+      return null;
+    }
+    try {
+      const decoded = this.decodeFunctionData({ data });
+      if (decoded.functionName !== 'deposit') {
+        throw new Error('Data is not of deposit type');
+      }
+      return {
+        method: decoded.functionName,
+        parameters: [],
+      };
```

Comment:
Nit: each decoder is method specific so I don't think we need expose the `functionName`/can just return the parameters. What do you think?

Reply:
After an offline sync, we decided to keep the `method` property in the return type. I've done so in [8fdaed7](https://github.com/safe-global/safe-client-gateway/pull/1903/commits/8fdaed7ac25cd5458e6c188bd282f238fd9ad3ba)